pennylane/resource/resource.py

```diff
@@ -29,21 +29,6 @@
 from .error import _compute_algo_error
 
 
-class SpecsDict(dict):
-    """A special dictionary for storing the specs of a circuit. Used to customize ``KeyError`` messages."""
-
-    def __getitem__(self, __k):
-        if __k == "num_diagonalizing_gates":
```

Comment:
I removed the warning associated with this since this specific field since it was removed back in March (several releases ago).